### ISSUE TITLE
Introduce an API to add entries

### ISSUE THREAD

**Issue by javydreamercsw** [mkalus/janag#2]
Right now everything is file based. Using a database could be better. I just don't know enough of it to implement it.

I use this in my game but I have to add entries to the files, compile, etc. With a database, for example, it could be dynamic.

**Comment by mkalus** [comment]
Possibly, we could introduce a generic data backend, e.g. file based, db, rest, another JaNaG server, etc. That would make the program more flexible. Question is, would it be possible to change the data during runtime - would it make sense to do that?

**Comment by javydreamercsw** [comment]
What I meant for runtime, was the ability of updating the data without the need to compile and restart like it is now.

**Comment by javydreamercsw** [comment]
You might have a service getting census data from somewhere so it's updated automatically,or some other source.

**Comment by mkalus** [comment]
Yes, that would make sense. A pluggable data source would be cool:

TODOs would be:
- define data source interface
- create default file based backend implementation of this interface
- add setting to define backend at startup of application

Now, other backends could be added...

**Comment by javydreamercsw** [comment]
I guess file backend would be default, but database comes to mind as well. As long as the interface is well defined, I could implement a database one using JPA.

**Comment by javydreamercsw** [comment]
Any updates or plans for this? I don't understand much of the engine to generate the names, so I can't be of much help until the interface is created and well documented.
